### PR TITLE
redisenterprise: refactoring to use the Embedded SDK

### PR DESCRIPTION
**Pull request by tombuildsstuff**

### CODE REVIEW

File: internal/services/redisenterprise/redis_enterprise_database_resource.go [hunk 1]
```diff
@@ -255,45 +258,50 @@ func resourceRedisEnterpriseDatabaseRead(d *pluginsdk.ResourceData, meta interfa
...
+		if props := model.Properties; props != nil {
+			d.Set("client_protocol", props.ClientProtocol)
+			d.Set("clustering_policy", props.ClusteringPolicy)
+			d.Set("eviction_policy", props.EvictionPolicy)
+			if err := d.Set("module", flattenArmDatabaseModuleArray(props.Modules)); err != nil {
+				return fmt.Errorf("setting `module`: %+v", err)
+			}
+			// if err := d.Set("persistence", flattenArmDatabasePersistence(props.Persistence)); err != nil {
+			// 	return fmt.Errorf("setting `persistence`: %+v", err)
+			// }
```
Comment (katbyte) [inline internal/services/redisenterprise/redis_enterprise_database_resource.go:296]: how come this is commented out?

**Review by katbyte — APPROVED** [review]
looks good aside from one comment 🚀

File: internal/services/redisenterprise/redis_enterprise_database_resource.go (hunk 1 above)
Comment (tombuildsstuff) [reply internal/services/redisenterprise/redis_enterprise_database_resource.go:296]: (as discussed offline) this is coming from the original data source - I think this is WIP from @WodansSon?

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2021-12-13 at 14 15 00

**Pull request merged**